src/arrayMethodJoin.js

```diff
@@ -4,8 +4,39 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const result = new Array(0);
+    let result1 = '';
```

Comment:
```suggestion
    let result = '';
```
Цей масив не потрібен. Достатньо пройтись циклом по this, і на кожній ітерації додати певні перевірки.
На кожній ітерації додавати до result this[i] та separator